core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -186,8 +185,8 @@ object Uri extends UriFunctions {
 
 
   private def renderParamsAndFragment(writer: Writer, p: Query, f: Option[Fragment]): writer.type = {
-    if (p.nonEmpty) writer << '?' << p
-    if (f.isDefined) writer << '#' << UrlCodingUtils.urlEncode(f.get, spaceIsPlus = false)
+    if (!p.isEmpty) writer << p
```

Comment:
@bryce-anderson Do you have any time to chat about this this weekend? I
think I can put in a good chunk of time on Saturday and at least try to get
my intentions squared away, answer questions, and incorporate feedback;
Sunday shouldn't be too bad either.
On Wed, Mar 16, 2016 at 11:50 Jean-Rémi Desjardins <notifications@github.com>
wrote:

> In core/src/main/scala/org/http4s/Uri.scala
> <https://github.com/http4s/http4s/pull/533#discussion_r56358995>:
>
> > @@ -186,8 +185,8 @@ object Uri extends UriFunctions {
> >
> >
> >    private def renderParamsAndFragment(writer: Writer, p: Query, f: Option[Fragment]): writer.type = {
> > -    if (p.nonEmpty) writer << '?' << p
> > -    if (f.isDefined) writer << '#' << UrlCodingUtils.urlEncode(f.get, spaceIsPlus = false)
> > +    if (!p.isEmpty) writer << p
>
> Ya, I think I have a preference for Option[Query] if that's possible, if
> not I would be curious why it's not possible.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/http4s/http4s/pull/533/files/783efc4064e0152fc6624d30b0afdc5831eec0d1#r56358995>
>